test/relayclient/GSNConfigurator.test.ts

```diff
@@ -0,0 +1,105 @@
+import { constants } from '../../src/common/Constants'
+import { DeploymentResult } from '../../src/cli/CommandsLogic'
+import { PrefixedHexString } from 'ethereumjs-tx'
+import ContractInteractor from '../../src/relayclient/ContractInteractor'
+
+const { assert, expect } = chai.use(chaiAsPromised)
+
+contract('client-configuration', () => {
+  let env: TestEnvironment
+  let deploymentResult: DeploymentResult
+  let paymasterAddress: PrefixedHexString
+  before(async () => {
+    env = await GsnTestEnvironment.startGsn('http://localhost:8544')
+    deploymentResult = env.deploymentResult
+    // deploymentResult = loadDeployment('./build/gsn')
+    paymasterAddress = deploymentResult.naivePaymasterAddress
+  })
+  describe('#resolveConfigurationGSN', () => {
+    describe('failures', () => {
+      it('should fail with no params', async () => {
+        // @ts-ignore
+        await expect(resolveConfigurationGSN()).to.eventually.rejectedWith(/Cannot read property/)
+      })
+
+      it('first arg not provider', async () => {
+        // @ts-ignore
+        await expect(resolveConfigurationGSN({})).to.eventually.rejectedWith(/First param is not a web3 provider/)
+      })
+      it('no paymaster in config', async () => {
+        await expect(resolveConfigurationGSN(web3.currentProvider, {}))
+          .to.eventually.rejectedWith('Cannot resolve GSN deployment without paymaster address')
+      })
+      it('no contract paymaster address ', async () => {
+        await expect(resolveConfigurationGSN(web3.currentProvider, { paymasterAddress: constants.ZERO_ADDRESS }))
+          .to.eventually.rejectedWith('no code at address ')
+      })
+
+      it('wrong contract paymaster address', async () => {
+        await expect(resolveConfigurationGSN(web3.currentProvider, { paymasterAddress: deploymentResult.stakeManagerAddress }))
+          .to.eventually.rejectedWith('Not a paymaster contract')
+      })
+
+      it('wrong contract paymaster version', async () => {
+        // instead of deploying a new paymaster with a different version, we make our client version older
+        // since resolveConfigurationGSN creates its own ContractInteractor, we have to hook the class to modify the version
+        // after it is created...
+
+        const saveCPM = ContractInteractor.prototype._createPaymaster
+        try {
+          ContractInteractor.prototype._createPaymaster = async function (addr) {
+            (this as any).versionManager.componentVersion = '1.0.0-old-client'
+            console.log('hooked _createPaymaster with version')
+            return await saveCPM.call(this, addr)
+          }
+
+          await expect(resolveConfigurationGSN(web3.currentProvider, { paymasterAddress }))
+            .to.eventually.rejectedWith(/Provided.*version.*is not supported/)
+        } finally {
+          ContractInteractor.prototype._createPaymaster = saveCPM
+        }
+      })
+    })
```

Comment:
Altering the prototype in the test does not smell good. Isn't there a better way?

Reply:
actually, this test shoiuld be disabled, since all our versioning tests are currently broken...